usmqe/web/tendrl/mainpage/tasks/models.py

```diff
@@ -0,0 +1,116 @@
+"""
+Common page model for tasks.
+"""
+
+
+from webstr.core import By, WebstrModel, PageElement, BaseWebElementHelper
+import webstr.patternfly.contentviews.models as contentviews
+import webstr.common.form.models as form
+
+LOCATION = "#/admin/tasks"
+
+
+class StatusIconHelper(BaseWebElementHelper):
+    """
+    StatusIcon helper (Selenium webelement wrapper).
+    Provides basic methods for manipulation with a task status icon.
+    """
+
+    @property
+    def value(self):
+        """
+        find status
+
+        Returns:
+            state of task
+        """
+        icon_class = self.get_attribute('class')
+        if 'pficon-error-circle-o' in icon_class:
+            return 'failed'
+        elif 'pficon-ok' in icon_class:
+            return 'finished'
+        elif 'pficon-warning-triangle-o' in icon_class:
+            return 'warning'
+        elif 'fa-spinner' in icon_class:
+            return 'new or processing'
+        else:
+            raise Exception('Unknown icon state')
+
+
+class StatusIcon(PageElement):
+    """
+    Page element for a status icon.
+    """
+    _helper = StatusIconHelper
+
+
+class TasksMenuModel(WebstrModel):
+    """
+    Tasks page top menu
+    """
+    header = PageElement(by=By.XPATH, locator="//h1[contains(text(),'Tasks')]")
+    filter_by = form.Select(
+        By.XPATH,
+        '//select[contains(@ng-model, "filterBy")]')
+    filter_input = form.TextInput(By.ID, 'filter')
+    from_input = form.TextInput(
+        By.XPATH,
+        '//div[@ng-model="taskCntrl.date.fromDate"]/input')
+    to_input = form.TextInput(
+        By.XPATH,
+        '//div[@ng-model="taskCntrl.date.toDate"]/input')
+
+
+class TasksItemModel(contentviews.ListViewRowModel):
+    """
+    An item (row) in a Tasks list.
+    """
+
+    def _instance_identifier(self):
+        """
+        instance identifier: line number - 1
+        """
+        return self._name - 1
+
+# Design: https://redhat.invisionapp.com/share/8N93NO7Q4
+    status_icon = StatusIcon(
+        by=By.XPATH,
+        locator="./div/i")
+    name_label = PageElement(
+        by=By.XPATH,
+        locator="./div[2]/a")
+    name = name_label
+    task_id = PageElement(
+        by=By.XPATH,
+        locator="./div[2]/p")
+    submitted = PageElement(
+        by=By.XPATH,
+        locator="./div[3]/p[2]")
+    status = PageElement(
+        by=By.XPATH,
+        locator="./div[4]/p")
+
+
+class TasksListModel(contentviews.ListViewModel):
+    """
+    Page model for list of tasks.
+    """
+
+
+class TaskDetailsModel(WebstrModel):
+    """ task details page model """
```

Comment:
Different commenting style than in the rest of the file.
"""
Task details page model.
"""